sdkproject/Assets/Mapbox/Unity/MeshGeneration/Data/UnityTile.cs

```diff
@@ -121,6 +121,11 @@ internal void Recycle()
 			HeightDataState = TilePropertyState.None;
 			VectorDataState = TilePropertyState.None;
 
+			OnHeightDataChanged = null;
```

Comment:
We can't null these out, right? The empty delegate prevents need for null check on registration.
Also, what's the reason for this? We reset state, already.